mypy/checker.py

```diff
@@ -3499,20 +3500,40 @@ def analyze_iterable_item_type(self, expr: Expression) -> Tuple[Type, Type]:
         """Analyse iterable expression and return iterator and iterator item types."""
```

Comment:
Let's explain how iterables are handled in CPython. Priorities and possibilities.